File: java/src/main/java/ie/hl7/cda/fhir/exception/FhirMappingException.java
package ie.hl7.cda.fhir.exception;

/**
 * Exception thrown when the CDA-to-FHIR mapping fails.
 */
public class FhirMappingException extends RuntimeException {

    /**
     * Constructs a FhirMappingException with the given message.
     *
     * @param message error description
     */
    public FhirMappingException(final String message) {
        super(message);
    }

    /**
     * Constructs a FhirMappingException with a message and a cause.
     *
     * @param message error description
     * @param cause   the underlying exception
     */
    public FhirMappingException(final String message, final Throwable cause) {
        super(message, cause);
    }
}

File: dotnet/src/CdaToFhirAdapter.Core/Exceptions/FhirMappingException.cs
namespace CdaToFhirAdapter.Core.Exceptions;

/// <summary>Exception raised when CDA-to-FHIR mapping fails.</summary>
public sealed class FhirMappingException : Exception
{
    /// <summary>Initializes a new instance with a message.</summary>
    public FhirMappingException(string message) : base(message) { }

    /// <summary>Initializes a new instance with a message and inner exception.</summary>
    public FhirMappingException(string message, Exception inner) : base(message, inner) { }
}
